File: golang/requester/fire-and-forget.go
package requester

import (
	"log"
	"github.com/rsocket/rsocket-go/payload"
)

//ExecFireAndForget ...
func ExecFireAndForget() {
	log.Println("====ExecFireAndForget====")
	cli, _ := BuildClient()
	defer cli.Close()
	cli.FireAndForget(payload.NewString("hello", "bonjour"))
}

File: rust/src/requester/fire_and_forget.rs
use futures::prelude::*;
use rsocket_rust::prelude::*;

use crate::requester::client;
pub fn exec_fire_and_forget() {
    println!("====ExecFireAndForget====");

    let cli = client::build_client();

    let fnf = Payload::from("Mock FNF");
    cli.fire_and_forget(fnf).wait().unwrap();

    let pa = Payload::builder()
        .set_metadata_utf8("metadata only!")
        .build();
    // metadata push
    cli.metadata_push(pa).wait().unwrap();

    let _ = cli.on_close();
}
